### PR TITLE
Custom component

### PR DESCRIPTION
**Pull request by hamishhossack**
# Enable custom components

The component rendered by the application when a auth route is currently reloading the page.

This new option will allow the user to customise this component, for example; If the component needs a specific layout.

### CODE REVIEW

File: lib/route.js [hunk 1]
```diff
@@ -1,6 +1,10 @@
 module.exports = {
-  mounted () {
-    window.location.reload()
+  layout: 'public',
```
Comment (samtgarson) [inline lib/route.js]: Don't think this should be here?

File: index.js [hunk 2]
```diff
@@ -5,7 +5,8 @@ const defaultOptions = {
   moduleName: 'oauth',
   fetchUser: () => ({}),
   onLogout: () => {},
-  scopes: []
+  scopes: [],
+  component: null
```
Comment (samtgarson) [inline index.js]: Could you name this something more specific, like `loadingPagePath` or something that a) specifies what the component is used for and b) specifies that it's a path, not an object or whatever

File: lib/route.js [hunk 3]
```diff
@@ -1,6 +1,10 @@
...
+  authenticated: false,
+  middleware () {
```
Comment (samtgarson) [inline lib/route.js]: Middleware receives `{ redirect }` which probably would be safer to use than `window.location`

File: lib/route.js (hunk 1 above)
Comment (hamishhossack) [reply lib/route.js]: no it shouldn't

File: lib/route.js (hunk 3 above)
Comment (hamishhossack) [reply lib/route.js]: It should be what was there before so it's backwards compatible, I just forgot to remove it ;)

**Pull request merged**